tests/src/App/ExpressionParser.cpp

```diff
@@ -0,0 +1,106 @@
+#include "gtest/gtest.h"
+
+#include "Base/Quantity.h"
+
+#include "App/Application.h"
+#include "App/Document.h"
+#include "App/DocumentObject.h"
+#include "App/Expression.h"
+#include "App/ExpressionParser.h"
+
+#include "src/App/InitApplication.h"
+
+// clang-format off
+
+class ExpressionParserTest: public ::testing::Test
+{
+protected:
+    static void SetUpTestSuite()
+    {
+        tests::initApplication();
+    }
+
+    void SetUp() override
+    {
+        _doc_name = App::GetApplication().getUniqueDocumentName("test");
+        _this_doc = App::GetApplication().newDocument(_doc_name.c_str(), "testUser");
+        _this_obj = _this_doc -> addObject("Sketcher::SketchObject");
+    }
+
+    void TearDown() override
+    {
+        App::GetApplication().closeDocument(_doc_name.c_str());
+    }
+
+    std::string doc_name() { return _doc_name; }
+    App::Document* this_doc() { return _this_doc; }
+    App::DocumentObject* this_obj() { return _this_obj; }
+
+    Base::Quantity parse_expra_as_quant(const char* expra_text) {
```

Comment:
Just a very small comment: why not use `expr` instead of `expra` anywhere in this file. (I'm trying to figure out what an `expra` could mean. I probably simply miss it, but to me the `a` suffix currently doesn't mean anything, so it confuses me.)